File: DSA/to_revise/1524 - numOfSubarrays.py
# https://leetcode.com/problems/number-of-sub-arrays-with-odd-sum/

from typing import List

class Solution:
    def numOfSubarrays(self, arr: List[int]) -> int:
        res = cur_sum = odd = even = 0
        MOD = 10 ** 9 + 7
        for n in arr:
            cur_sum += n
            if cur_sum % 2:
                res = (res + 1 + even) % MOD
                odd += 1
            else:
                res = (res + odd) % MOD
                even += 1
        return res

File: DSA/to_revise/1524 - numOfSubarrays.cpp
// https://leetcode.com/problems/number-of-sub-arrays-with-odd-sum/

#include <vector>
using namespace std;

class Solution {
    public:
        int numOfSubarrays(vector<int>& arr) {
            int curSum(0), odd(0), even(0), res(0);
            const int MOD = 1e9 + 7;
    
            for (auto n : arr) {
                curSum += n;
                if (curSum % 2) {
                    res = (res + 1 + even) % MOD;
                    odd++;
                } else {
                    res = (res + odd) % MOD;
                    even++;
                }
            }
    
            return res;
        }
    };